equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/repository/criteria/input/CrmFormDetailsCriteriaDTO.java

```diff
@@ -12,6 +13,8 @@ public record CrmFormDetailsCriteriaDTO(
         if (usn == null) {
             throw new InvalidDataAccessApiUsageException("Expected USN not be null");
         }
+
+        ProfileAcceptedTypesUtil.checkTypeIsAcceptedByProfile(type, profileAcceptedTypes);
```

Comment:
method name suggestion - `isAcceptedProfileType` or `isProfileTypeAccepted`

Reply:
I am not using the is prefix because this method is not returning. I use that convention for boolean returns. In this case, the check will raise an exception when not passed

Reply:
maybe change to `checkTypeIsAcceptedForProfile()`